### ISSUE TITLE
Use Danger to simplify PR reviews

### ISSUE THREAD

**Issue by ananyo2012** [publiclab/plots2#1192]
[Danger](http://danger.systems/) is a good way to implement automated PR reviews. Now that we have growing no of contributors in our codebase it is good to have automated reviews so it is easier to do reviews as well for contributors.

**Comment by jywarren** [comment]
Hi, sounds great -- i'd love to be able to:

* ask people who haven't done anything in 4+ days if they need help, and offer to ping `@publiclab/reviewers`
* see if people have merged from master and request that they not do so (linking to https://publiclab.org/wiki/contributing-to-public-lab-software#Rewinding+the+master+branch)
* see if people's tests have passed and show how to look through the logs to find the error. Or even automatically excerpt in the error?
* ask people who've just closed a `first-timers-only` issue if they're looking to contribute more, and link them to `fto-candidates` with a helpful message
* ask people who've completed a second project if they'd like to look at harder `help-wanted` issues

Many more! This would be great. And I think @ryzokuken has ideas too.

**Comment by ananyo2012** [comment]
Well Danger may not fulfill all these things. But I would surely try to implement as many as possible. I thing we should update the PR template too reflect the changes. The second last point that mentions `reviewed/confirmed/tested by another contributor or maintainer` can be removed since github now has this review feature and reviewers can approve/reject from there.

**Comment by jywarren** [comment]
Just listing my blue-sky ideas in an ideal world -- and I don't know what
Danger is capable of, but any of these would be helpful!

Happy to accept PRs for the PR template itself too!

**Comment by jywarren** [comment]
Oh, also:

* look for and request (if not present) a link to the originating issue

**Comment by ananyo2012** [comment]
- Check for linting errors in PR

**Comment by yukiisbored** [comment]
I think gitmate.io is better than Danger because it's much simpler and it's all we need. gitmate.io runs coala and reports back the reported linter issue and I think it doesn't need to be invoked from CI since it's a Webhook.

**Comment by ananyo2012** [comment]
@yukiisbored Well I saw gitmate is still in beta and I didn't find a good documentation for gitmate. Danger seemed well documented and I really liked the plugin feature through which we can write our own custom plugins as per our requirement. Maybe we can switch to gitmate once it is complete and has its all features.

**Comment by jywarren** [comment]
Yes, I think we can stay flexible as gitmate.io develops and work with Danger as a starting point. Also, Ananyo's done a lot of the work to get Danger running already!

**Comment by yukiisbored** [comment]
@ananyo2012 Can you add an issue to https://gitlab.com/gitmate/gitmate-2 for that?
